src/tracks/Pattern.cpp

```diff
@@ -870,23 +875,24 @@ void PatternView::paintEvent( QPaintEvent * )
 
 	setNeedsUpdate( false );
 
-	m_paintPixmap = m_paintPixmap.isNull() == true || m_paintPixmap.size() != size()
-		? QPixmap( size() ) : m_paintPixmap;
+	if (m_paintPixmap.isNull() || m_paintPixmap.size() != size())
```

Comment:
This is a much cleaner than the previous approach. We do this logic for all the other TCO's, can you update that logic too?